### ISSUE TITLE
Handling errors without domains (domains deprecated)

### ISSUE THREAD

**Issue by runvnc** [Automattic/kue#904]
Readme says the best way to handle errors is by using domains.  But Node.js docs says domains are going away.  So is there another way that will actually work?

**Comment by behrad** [comment]
You're right @runvnc , Best practices are as general Node's error handling ones.
We should update docs.

**Comment by runvnc** [comment]
Is it possible to catch and attribute all errors and fail jobs that caused them, with done(), without using domains?

**Comment by behrad** [comment]
> Is it possible to catch and attribute all errors and fail jobs that caused them, with done(), without using domains?

Yes, https://github.com/CrabDude/trycatch can help you to do this :)

**Comment by runvnc** [comment]
LOL. I don't want to take up too much of your time with this.  But trycatch
is based on domains.. so if they remove domains from Node then that won't
be an option either.  So if we really need one of those to make sure we
fail jobs, then we better hope they don't actually remove domains..which I
don't see a problem with domains at all, just had to ask because the docs
state that they are going to remove them.

Maybe I should ask the Node.js maintainers what kue users are supposed to
do if they take out domains.

**Comment by behrad** [comment]
> But trycatch
> is based on domains..

Then you can opt for Promise modules error handling facilities :) & Read carefully how to handle sync/async errors with them.
